src/proposer/proposer_rpc.cpp

```diff
@@ -167,6 +170,134 @@ class ProposerRPCImpl : public ProposerRPC {
     obj.pushKV("stakeable_coins", arr);
     return obj;
   }
+
+  UniValue PorposeBlocks(CWallet *const wallet,
+                         const boost::optional<CScript> &coinbase_script,
+                         int num_generate) const {
+    assert(wallet);
+    blockchain::Height height = 0;
+    blockchain::Height height_end = 0;
+
+    proposer::Proposer *proposer = GetComponent<proposer::Proposer>();
+
+    staking::ActiveChain *active_chain;
+    {  // Don't keep cs_main locked
+      LOCK(cs_main);
+      active_chain = GetComponent<staking::ActiveChain>();
+      height = active_chain->GetHeight();
+      height_end = height + num_generate;
+    }
+    UniValue block_hashes(UniValue::VARR);
+
+    // To pick up to date coins for staking we need to make sure that the wallet is synced to the current chain.
+    wallet->BlockUntilSyncedToCurrentChain();
+
+    esperanza::WalletExtension &wallet_ext = wallet->GetWalletExtension();
+
+    while (height < height_end) {
+
+      std::shared_ptr<const CBlock> block;
+      {
+        LOCK2(cs_main, wallet_ext.GetLock());
+        const staking::CoinSet stakeable_coins = wallet_ext.GetStakeableCoins();
+
+        if (stakeable_coins.empty()) {
+          throw JSONRPCError(RPC_INTERNAL_ERROR,
+                             "Not proposing, not enough balance.");
+        }
+        staking::CoinSet coins;
+        coins.insert(*stakeable_coins.begin());
+
+        if (!proposer->GenerateBlock(wallet,
+                                     *active_chain->GetTip(),
+                                     coins,
+                                     coinbase_script,
+                                     block)) {
+          throw JSONRPCError(RPC_INTERNAL_ERROR, "Failed to generate a block.");
+        }
+      }
+
+      if (!ProcessNewBlock(Params(), block, /* fForceProcessing= */ true, nullptr)) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "ProcessNewBlock, block not accepted");
+      }
+
+      ++height;
+      block_hashes.push_back(block->GetHash().GetHex());
+
+      wallet->BlockUntilSyncedToCurrentChain();
+    }
+    return block_hashes;
+  }
+
+  UniValue propose(const JSONRPCRequest &request) const {
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet *const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+      return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
```

Comment:
this function seems to accept only 1 parameter however it will execute (not render the help) when 2 are passed.